contracts/erc721/PaymentObligation.sol

```diff
@@ -127,47 +126,73 @@ contract PaymentObligation is Initializable, UserMintableERC721 {
       anchorId
     );
 
+
+    // Check if status of invoice is unpaid
+    require(
+      MerkleProof.verifySha256(
+        proofs[4],
+        merkleRoot_,
+        sha256(
+          abi.encodePacked(
+            hex"0001000000000002", // compact property for  invoice.status, invoice = 1, status = 2
+            hex"756e70616964", // bytes value for "unpaid"
+            salts[4]
+          )
+        )
+      ),
+      "Invoice status is not unpaid"
+    );
+
+    // Check if sender is a registered identity
+    super._requireValidIdentity(
+      merkleRoot_,
+      hex"0001000000000013", // compact property for  invoice.sender, invoice = 1, sender = 19
+      values[3],
+      salts[3],
+      proofs[3]
+    );
+
     // Enforce that there is not a newer version of the document on chain
     super._requireIsLatestDocumentVersion(
       merkleRoot_,
       nextAnchorId,
-      salts[3],
-      proofs[3]
+      salts[5],
+      proofs[5]
     );
 
     // Verify that only one token per document/registry is minted
     super._requireOneTokenPerDocument(
       merkleRoot_,
       tokenId,
-      salts[4],
-      proofs[4]
+      salts[6],
+      proofs[6]
     );
 
     // Check if document has a read rule defined
     bytes8 readRoleIndex = super._requireReadRole(
       merkleRoot_,
       properties[0],
-      values[3],
-      salts[5],
-      proofs[5]
+      values[4],
+      salts[7],
```

Comment:
I would like to have constants for everything. No indexing with numbers it unnecessary complicated to read in my opinion.